pandas/tools/tile.py

```diff
@@ -81,6 +83,17 @@ def cut(x, bins, right=True, labels=None, retbins=False, precision=3,
     array([1, 1, 1, 1, 1], dtype=int64)
     """
     # NOTE: this binning code is changed a bit from histogram for var(x) == 0
+    # for handling the cut for datetime and timedelta objects
+
```

Comment:
Yes, I agree preprocessing/postprocessing should be moved to ``cut/qcut`` from ``_bins_to_cuts``. what on my mind is something like below. 

Actually preprocessing / postprocessing can be wrapped as a private function to reduce code duplications. 

```
def cut(...):
    x_is_series = ...
    ... 
    x = np.asarray(x)    
    if is_datetime64_dtype(...):
        ....

    (core logic)

    if x_is_series:
        ....
    return...
```